### PR TITLE
[CSDP] Version 6.2.0

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

**Comment by ViralBShah** [comment]
Just a quick pointer: https://github.com/JuliaPackaging/Yggdrasil/pull/796

This is how I am organizing the various Coin-OR solvers. It would be nice to have CSDP use the same framework. Can be done later once it's all working. At least it doesn't have major dependencies on the rest of Coin-OR, and looks fairly standalone.

**Comment by giordano** [comment]
Note that we expect the executables to have the appropriate extension (`${exeext}`) for each platforms:

* `${exeext}` = `".exe"` for Windows
* `${exeext}` = `""` (empty string) elsewhere

The `x86_64-w64-mingw32` build is actually successful, but the executable is called `csdp`, without the expected extension.  You should add the end of the build script something like
```sh
if [[ "${target}" == *-mingw* ]]; then
    mv "${bindir}/csdp" "${bindir}/csdp${exeext}"
fi
```

**Comment by amontoison** [comment]
Thanks @giordano ! I also remarked that `bindir` folder doesn't exist, we need to create it ourself which is not the case of `libdir`.

**Comment by amontoison** [comment]
I moved everything in Coin-OR folder @ViralBShah.

File: C/Coin-OR/CSDP/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency("OpenBLAS32_jll"),
+    Dependency("CompilerSupportLibraries_jll"),
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
```
Comment (giordano) [inline C/Coin-OR/CSDP/build_tarballs.jl:68]: I think you may need to use GCC v6 here:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version=v"6")
```

**Comment by ViralBShah** [comment]
No shared libraries built on this package?

**Comment by ViralBShah** [comment]
Let's disable powerpc. It doesn't work for any of the libraries in coin-or folder.

**Comment by ViralBShah** [comment]
Shouldn't there be a `libcsdp` LibraryProduct produced?

**Comment by amontoison** [comment]
Only 5 executables are producted at the end `csdp`, `theta`, `graphtoprob` `complement` and `rand_graph`. There isn't `LibraryProduct` for this package.

Did you see if the compilation worked on PowerPC platform with gcc `v6`?

**Comment by ViralBShah** [comment]
I thought I saw a libcdsp.a. Would it be useful as a shared library?

**Comment by amontoison** [comment]
It wasn't used inside https://github.com/JuliaOpt/CSDP.jl, I don't know if it is be useful.

**Comment by blegat** [comment]
> I thought I saw a libcdsp.a. Would it be useful as a shared library?

CSDP.jl requires a CSDP shared library. It does not used the CSDP executable.
As the makefile only builds an executable and a static library, The current way the shared library is built is as follows:
https://github.com/JuliaOpt/CSDP.jl/blob/958078c0744293be2a3179f20052ccb85e3a592b/deps/compile.jl#L64-L98
This is a bit hacky and we are of course open to a better solution.

Note that the `path_int` function:
https://github.com/JuliaOpt/CSDP.jl/blob/958078c0744293be2a3179f20052ccb85e3a592b/deps/compile.jl#L31-L61
is needed when we link to the openblas library shipped with Julia with the `_64` suffix but it does not seem to be the case here so you can just ignore it.

**Comment by giordano** [comment]
We can get a shared library out of the static archive if necessary. We have to do this very often

**Comment by ViralBShah** [comment]
Can you point to an example of how to do that?

**Comment by giordano** [comment]
It's basically something like
```sh
# The static archives are usually in the lib/ directory,
# also when building for Windows
cd "${prefix}/lib"
# Extract the content of the static archive
ar x libfoo.a
# Compile all the object files into a the shared library.
# Use the appropriate compiler (`cc` for C, `c++` for C++, etc...).
# You may need to pass some additional flags for the linker
cc -shared -o "${libdir}/libfoo.${dlext}" *.o
# Remove all the object files
rm *.o
```

**Comment by ViralBShah** [comment]
Ah ok -  I thought BB had something magical like `SharedfromStatic` or something :-)

**Comment by giordano** [comment]
Not yet :joy:

**Pull request merged**